### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by robertkowalski**
this implements bitfinex terminal stream processing with examples.
it includes a helper method for submitting data to paid or free
dazaar feeds.

if a submission to a websocket is required, it can be easily build
using a custom `submitOrder` function.

examples for both cases are included

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,100 @@
...
+  symbol: SYMBOLS.BTC_USD,
+  tf: TIME_FRAMES.ONE_HOUR
+}
+
+const strategy = EMAStrategy(market)
+const submitOrder = util.getSubmitOrderToFeed(sellFeed)
+
+const { exec, stream } = await execDazaar(strategy, market, db, {
+  submitOrder,
+
```
Comment (mafintosh) [inline README.md]: nit: extra space

File: README.md [hunk 2]
```diff
@@ -1 +1,100 @@
...
+  includeTrades: false,
+  seedCandleCount: 10
+})
+
+stream.on('data', async (data) => {
```
Comment (mafintosh) [inline README.md]: we cannot async iterate this stream?

File: README.md [hunk 3]
```diff
@@ -1 +1,100 @@
...
+  const { key, value } = data
+
+  btState = await exec(key, value)
+})
+```
+
+#### Trade with Custom Order Submit Functions
+
+You can use any custom function to handle trading signals, as lang as it returns a promise.
```
Comment (mafintosh) [inline README.md]: * as long

File: README.md [hunk 4]
```diff
@@ -1 +1,100 @@
...
+    type,
+    price
+  }, ws)
+
+  await o.submit()
+})
+
+const { exec, stream } = await execDazaar(strategy, market, db, {
+  submitOrder,
+
```
Comment (mafintosh) [inline README.md]: extra newline

File: package.json [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    "dazaar"
+  ],
+  "dependencies": {
+    "bfx-api-node-models": "^1.2.11",
+    "bfx-api-node-util": "^1.0.9",
+    "bfx-hf-strategy": "^1.0.5",
+    "debug": "^4.1.1"
+  },
+  "devDependencies": {
+    "@dazaar/payment": "^1.0.2",
```
Comment (JacobPlaster) [inline package.json:39]: A lot of these shouldn't be in devDeps right?

**Comment by JacobPlaster** [comment]
Lets add an Apache 2.0 license: https://github.com/bitfinexcom/bfx-hf-algo-server/blob/master/LICENSE

File: package.json (hunk 5 above)
Comment (robertkowalski) [reply package.json:39]: they are just used for the example-code, for the main lib just debug, bfx-hf-strategy, bfx-api-node-models, bfx-api-node-util are required

(in the files index.js and util.js)

**Pull request merged**